roletracker/roletracker.py

```diff
@@ -0,0 +1,253 @@
+
+        if not await self.config.role(role).addable():
+            return await ctx.maybe_send_embed("Role isn't set as removable.")
+        guild = ctx.guild
+        try:
+            if role not in member.roles:
+                return await ctx.maybe_send_embed("Member doesn't have that role.")
+
+            data = await self.config.role(role).USERS()
+            caseno = data.pop(member.id, None)
+
+            if caseno:
+                try:
+                    case = await modlog.get_case(caseno, guild, self.bot)
+                except RuntimeError:
+                    self.logger.error(f"Failed to find case for {member}, case number: {case}")
+                    case = None
+
+                if case:
+                    edits = {"reason": reason}
+
+                    if ctx.message.author.id != case.moderator.id:
+                        edits["amended_by"] = ctx.message.author
+
+                    edits["modified_at"] = ctx.message.created_at.timestamp()
+
+                    await case.edit(edits)
+
+            await member.remove_role(role)
+            await self.config.role(role).USERS.set(data)
+            await ctx.tick()
+        except discord.Forbidden:
+            return await ctx.maybe_send_embed("Can't do that. Discord role heirarchy applies here.")
+
+    # Listeners
+    @commands.Cog.listener()
+    async def on_member_update(self, before: discord.Member, after: discord.Member):
+        """
+        Listens for role updates and more
+        """
+
+        if before.roles != after.roles:
+            user = self.bot.user
+            try:
+                async for entry in after.guild.audit_logs(limit=3, action=discord.AuditLogAction.member_role_update):
+                    if entry.target.id == before.id:
+                        user = entry.user
+                        break
+            except discord.Forbidden:
+                self.logger.warning("Failed to retrieve the moderator from audit logs, please check the permissions")
+
+            broles = set(before.roles)
+            aroles = set(after.roles)
+            added = aroles - broles
+            removed = broles - aroles
+
+            now_date = datetime.utcnow()
+
+            for role in added:
+                if await self.config.role(role).addable():
+                    data = await self.config.role(role).USERS()
+
+                    case = await modlog.create_case(
+                        self.bot,
+                        before.guild,
+                        now_date,
+                        "Role List Update",
+                        before,
+                        moderator=user,
+                        reason="Role manually added",
+                    )
+                    caseno = case.case_number
+
+                    data[before.id] = caseno
+                    await self.config.role(role).USERS.set(data)
+
+            for role in removed:
+                if await self.config.role(role).addable():
```

Comment:
same as above for removed command, need to check if it was done with command and not create dupelicate cases.